docs/advanced-usage/using-proxies.mdx

```diff
@@ -0,0 +1,225 @@
+---
+title: Using Proxies
+description: Learn how to proxy the Clerk Frontend API through your domain.
+---
+
+# Proxying the Clerk Frontend API
+
+Clerk supports two configuration methods for connecting to the Clerk Frontend API: CNAME and Proxy
+
+[Setting up CNAME records and using DNS is the simplest way](/docs/deployments/overview). However, if you're unable to use this approach, or would like more control over your integration with Clerk, you can use a proxy.
+
+When using a proxy, all requests to the Frontend API will be made through your domain. This allows you to use your own SSL certificate, and gives you more control over how you configure your application.
+
+## How to use Proxying
+
+<Steps>
+### Create your application and install Clerk
+
+To get started, you need to create an Application from the [Clerk Dashboard](https://dashboard.clerk.com/). Once you create an instance via the Clerk Dashboard, you will be prompted to choose a domain. For the purposes of this guide, the domain will be `app.dev`.
+
+<Callout type="warning">Proxying is a replacement for the CNAME approach. As such, if you're using a proxy, you cannot use CNAME.</Callout>
+
+### Configure your proxy server
+
+For this example, we've chosen `/__clerk` as the path for our proxy. Your proxy server must be on the same domain as your application. 
+
+You can choose any path you'd like, but it must be unique and not conflict with any other routes in your application.
+
+#### Requirements
+
+Requests to `https://app.dev/__clerk/*` must be forwarded to `https://frontend-api.clerk.dev/*` with the body and all headers intact.
+
+##### Three additional headers must be set:
+
+  - `Clerk-Proxy-Url`: Needs to have the full proxy URL.
+  - `Clerk-Secret-Key`: The secret key for your Clerk instance.
+  - `X-Forwarded-For`: The IP address of the original client making the request.
+
+#### Example Configuration
+
+<Tabs items={['Nginx', 'Cloudflare Workers']}>
+  <Tab>
+    ```conf filename="nginx.conf"
+    http {
+      # ...
+      server {
+        # ...
+        location /__clerk/ {
+          rewrite ^/__clerk/(.*)$ /$1 break;
+          proxy_pass https://frontend-api.clerk.dev;
+          proxy_set_header Clerk-Proxy-Url https://app.dev/__clerk;
+          proxy_set_header Clerk-Secret-Key sk_live_***;
```

Comment:
we are missing `proxy_set_header X-Forwarded-For $remote_addr;` here.

Reply:
Great catch! Fixed in https://github.com/clerkinc/clerk-docs/pull/388/commits/edd79b2ea3a0630c64a72b2d1d1d225f942bd74e